src/Filter/Filter.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types=1);
+
+namespace swentel\nostr\Filter;
+
+use swentel\nostr\FilterInterface;
+
+class Filter implements FilterInterface
+{
+    /**
+     * A list of event ids
+     */
+    public array $id;
+
+    /**
+     * A list of lowercase pubkeys, the pubkey of an event must be one of these
+     */
+    public array $authors;
+
+    /**
+     * A list of a kind numbers
+     */
+    public array $kinds;
+
+    /**
+     * A list of #e tag values (list of event ids)
+     */
+    public array $etags;
+
+    /**
+     * A list of #p tag values (list of pubkeys).
+     */
+    public array $ptags;
```

Comment:
FYI Nostr event can have lowercase and uppercase p tags (zap receipt kind `9735` event for example)